pandas/core/indexing.py

```diff
@@ -2024,10 +2024,18 @@ def _setitem_single_column(self, loc: int, value, plane_indexer) -> None:
 
         is_null_setter = com.is_empty_slice(pi) or is_array_like(pi) and len(pi) == 0
 
+        is_empty_df_setter = com.is_null_slice(pi) and self.obj._mgr.arrays[0].size == 0
```

Comment:
any reason for this to be `self.obj._mgr.arrays[0].size == 0` instead of `len(self.obj) == 0`?